apollo-router/src/router_factory.rs

```diff
@@ -765,6 +765,15 @@ pub(crate) async fn create_plugins(
     add_optional_apollo_plugin!("coprocessor");
     add_user_plugins!();
 
+    if apollo_plugin_factories.contains_key("apollo.subgraph_mocks") {
+        // Defined in apollo-router/tests/integration/subgraph_mocks_plugin/mod.rs,
+        // only present when running that test executable.
+        // Needs to come after user plugins
+        // but before `__internal_for_test_harness_prevent_all_subgraph_requests`.
+        add_optional_apollo_plugin!("subgraph_mocks");
+    }
+    add_optional_apollo_plugin!("__internal_for_test_harness_prevent_all_subgraph_requests");
```

Comment:
I’m not really happy with hard-coding so much here but I don’t know how else to manage plugin ordering.

Going through `TestHarness::extra_plugin` instead of `register_plugin!` does give control over ordering, but requires an already-instanciated plugin which didn’t go through `Plugin::new` and did get access to `PluginInit` and through it subgraph schemas. Exfiltrating subgraph schemas from `TestHarness` does not look easy since there are many layers of nested types (including a `Box<dyn Service>`)